blocks/api/serializer.js

```diff
@@ -165,11 +172,14 @@ export function getBlockContent( block ) {
 	const blockType = getBlockType( block.name );
 
 	// If block was parsed as invalid or encounters an error while generating
-	// save content, use original content instead to avoid content loss.
+	// save content, use original content instead to avoid content loss. If a
+	// block contains nested content, exempt it from this condition because we
+	// otherwise have no access to its original content and content loss would
+	// still occur.
 	let saveContent = block.originalContent;
-	if ( block.isValid ) {
+	if ( block.isValid || block.innerBlocks.length ) {
```

Comment:
So, If I understand properly we're ignoring the validation because we can't validate properly the container blocks? Can you explain a bit more why we can't validate the container blocks properly? (I'm certain you already did elsewhere, just a link maybe :))

Why are we creating a difference between blocks with inner blocks and blocks without inner blocks. If we can't guarantee the behavior for all blocks, shouldn't we just remove the condition to make it consistent?

Reply:
The main issue is that if we mark the block as invalid, it will defer the next save to using its `innerHTML`; this is usually meant to preserve the original content, but in a case of a block which contains nested blocks, it destroys the content, because nested content is not reflected in a block's `innerHTML`.

Reply:
Also, that we'd be validating the same content n-times.